src/main/java/org/cryptomator/cloudaccess/MetadataCachingProviderDecorator.java

```diff
@@ -70,94 +70,98 @@ public MetadataCachingProviderDecorator(CloudProvider delegate, Duration cacheEn
 
 	@Override
 	public CompletionStage<CloudItemList> list(CloudPath folder, Optional<String> pageToken) {
-		return delegate.list(folder, pageToken) //
-				.whenComplete((cloudItemList, exception) -> {
-					evictIncludingDescendants(folder);
-					if (exception == null) {
-						assert cloudItemList != null;
-						cloudItemList.getItems().forEach(metadata -> itemMetadataCache.put(metadata.getPath(), CompletableFuture.completedFuture(metadata)));
-					}
-				});
+		try {
+			var folderKey = Map.entry(folder, pageToken);
```

Comment:
I'd suggest using a record. Misusing `Map.Entry` to fix Java's lack of tuples feels wrong.

Reply:
And the type declarations for the Cache are way too long.